flow_action_components/MultipicklistValues/force-app/main/default/classes/MultiPicklistAdd.cls

```diff
@@ -0,0 +1,49 @@
+global class MultiPicklistAdd {
+  @InvocableMethod(
+    label='Add Multi-Picklist Val(s)'
+    description='Given A multi-picklist value (as string), will add the target the value(s) from it and return a new picklist value (as string)'
+    category='Picklist'
+  )
+  global static List<String> addPicklistVal(
+    List<MultiPicklistAddParams> params
```

Comment:
Two comments on your approach to inputs:

1) the standard pattern we use is to define Request and Response classes and pass those in and out of the invocable action, instead of directly passing data like you do here. This enables bulkification to work automatically: if someone triggers a flow on a batch of 40 records and your action is in that flow, it will not automatically and properly bulkify your current code because it doesn't have the empty outer list that Request and Response provide. For examples, see the String Changers stuff. While this isn't a breaking problem, I'm motivated to keep consistency unless there's a good reason.

You're making your input an Apex-Defined type that contains two strings. That's ok, but in this case, I would think you'd just want to put the two strings independently into the 'Request' class. That way they show up as individual strings in Flow Builder.

Reply:
Can you add as comments in the code some examples of inputs and the resulting outputs?

Reply:
I'll take a look at the other string changers and refactor to make.  I'll try to get around to it this week (as well as wrap up the split/join/unique stuff)